src/runtime-rs/crates/service/src/manager.rs

```diff
@@ -5,30 +5,91 @@
 //
 
 use std::{
+    fs,
     os::unix::io::{FromRawFd, RawFd},
+    process::Stdio,
     sync::Arc,
 };
 
 use anyhow::{Context, Result};
-use common::message::{Action, Message};
-use containerd_shim_protos::shim_async;
+use common::message::{Action, Event, Message};
+use containerd_shim_protos::{
+    protobuf::{well_known_types::Any, Message as ProtobufMessage},
+    shim_async,
+};
 use runtimes::RuntimeHandlerManager;
-use tokio::sync::mpsc::{channel, Receiver};
+use tokio::{
+    io::AsyncWriteExt,
+    process::Command,
+    sync::mpsc::{channel, Receiver},
+};
 use ttrpc::asynchronous::Server;
 
 use crate::task_service::TaskService;
 
 /// message buffer size
 const MESSAGE_BUFFER_SIZE: usize = 8;
 
+pub const KATA_PATH: &str = "/run/kata";
+
 pub struct ServiceManager {
     receiver: Option<Receiver<Message>>,
     handler: Arc<RuntimeHandlerManager>,
     task_server: Option<Server>,
+    containerd_binary: String,
```

Comment:
Note: also CRI-O should be considered.

Reply:
Thanks for your review, has been updated.